libraries/AP_Scripting/generator/src/main.c

```diff
@@ -608,6 +608,19 @@ int parse_type(struct type *type, const uint32_t restrictions, enum range_check_
     sanatize_name(&(type->data.ud.sanatized_name), type->data.ud.name);
   }
 
+  // only allow no range check on float, int32 and uint32
+  if (type->flags & TYPE_FLAGS_NO_RANGE_CHECK) {
+    switch (type->type) {
```

Comment:
why can't we auto-check for 0 to UINT16_MAX when skip_check is used?

Reply:
We could do that, or add a new "full_range" keyword. It makes no difference to the generated code, but might make the descriptions a little easier I guess.